packages/react-ui-ag/src/Listings/SingleFamilyMobileListing.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import autobind from 'autobind-decorator'
+import { ListingComponents } from '@rentpath/react-ui-core'
+import Listing from './Listing'
+
+const buttonPropTypes = PropTypes.shape({
+  children: PropTypes.node,
+  onClick: PropTypes.func,
+  className: PropTypes.string,
+  valueLocation: PropTypes.string,
+})
+
+@themed(['SingleFamilyMobileListing'])
+export default class SingleFamilyMobileListing extends Component {
+  static propTypes = {
+    index: PropTypes.number,
+    listing: PropTypes.object,
+    theme: PropTypes.object,
+    className: PropTypes.string,
+    onClick: PropTypes.func,
+    photos: PropTypes.object,
+    ratings: PropTypes.object,
+    propertyName: PropTypes.object,
+    ctaButtons: PropTypes.arrayOf(buttonPropTypes),
+    favoriteButton: buttonPropTypes,
+    lazyLoad: PropTypes.oneOfType([
+      PropTypes.object,
+      PropTypes.bool,
+    ]),
+    isActive: PropTypes.bool,
+    singleFamily: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    isActive: true,
+    lazyLoad: true,
+    theme: {},
+    listing: {},
+    ratings: {},
+    photos: {},
+  }
+
+  @autobind
+  getListingInfoComponents() {
+    return (
+      <React.Fragment>
+        <ListingComponents.Price />
+        <ListingComponents.Address />
+        <ListingComponents.Bedroom />
+        <ListingComponents.Availability />
+      </React.Fragment>
+    )
+  }
+
+  render() {
+    const { theme } = this.props
+
+    return (
+      <Listing
+        {...this.props}
+        className={theme.SingleFamilyMobileListing}
+        listingInfoComponents={this.getListingInfoComponents}
```

Comment:
Agree with @grenti. This is something we actually talked about a while ago.

Reply:
@grenti I did think about combining SingleFamily listings with the `MobileListing` and `DesktopListing` components. It seemed like it would lead to many `singleFamily` conditional checks inside of `getListingInfoComponents` if new features were to be added. My approach of putting single family code in a different component was to make it less messy and somewhat maintainable in the future. However, since there are a few differences between the Single Family and regular listings and there is repeated code throughout, we could definitely combine it into `DesktopListing` and `MobileListing`.

Actually, the `ListingInfo` component is a good idea too. Or we can create two separate functions in Desktop/Mobile Listing called `getRegularListingInfo` and `getSingleFamilyListingInfo` that would appropriately get called depending on `singleFamily` prop.

Reply:
@okhanrt I think the ListingInfo would be a good start so that it encapsulates the conditional and the components.

Reply:
@grenti sounds like a plan. Thanks!